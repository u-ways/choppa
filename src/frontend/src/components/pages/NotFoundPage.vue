<template>
  <div class="d-flex justify-content-center align-items-center p-2 h-100">
    <div class="d-flex flex-column">
      <div class="d-flex flex-row justify-content-center font-weight-bold mt-2">
        <ChoppaLogoAtom css="h1" />
      </div>
      <div class="d-flex flex-row">
        <h1 class="h5 justify-content-center mt-2 mb-0">
          404 - You've taken the wrong flight!
        </h1>
      </div>
      <div class="d-flex flex-row justify-content-center">
        <router-link to="/" class="d-flex mt-1">
          Fly home
        </router-link>
      </div>
    </div>
  </div>
</template>

<script>
import ChoppaLogoAtom from "@/components/atoms/ChoppaLogoAtom";

export default {
  name: "NotFoundPage",
  components: { ChoppaLogoAtom },
};
</script>

<style scoped lang="scss">
.icon {
  font-size: 4rem;
  text-align: center;
  width: 100%;
}
</style>
