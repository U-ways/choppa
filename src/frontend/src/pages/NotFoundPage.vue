<template>
  <NoticePageTemplate>
    <div class="flex flex-col">
      <h1 class="text-2xl text-center">404 - You've taken the wrong flight!</h1>
      <span class="text-center"><StyledLink link="/">Edit Tribe</StyledLink></span>
    </div>
  </NoticePageTemplate>
</template>

<script>
import NoticePageTemplate from "@/components/templates/NoticePageTemplate";
import StyledLink from "@/components/atoms/StyledLink";

export default {
  name: "NotFoundPage",
  components: {
    NoticePageTemplate,
    StyledLink,
  },
};
</script>
