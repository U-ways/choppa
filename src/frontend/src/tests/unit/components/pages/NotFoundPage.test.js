import { createLocalVue, describe, expect, it, shallowMount } from "@/tests/unit/test-imports";
import VueRouter from "vue-router";
import NotFoundPage from "@/_old/components/pages/NotFoundPage";

describe("Not Found Page test", () => {
  it("Renders correctly", async () => {
    const localVue = createLocalVue();
    localVue.use(VueRouter);
    const router = new VueRouter();

    const wrapper = shallowMount(NotFoundPage, {
      localVue,
      router,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
