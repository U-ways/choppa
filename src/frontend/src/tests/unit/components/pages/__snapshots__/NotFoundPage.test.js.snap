// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not Found Page test Renders correctly 1`] = `
"<div class=\\"d-flex justify-content-center align-items-center p-2 h-100\\">
  <div class=\\"d-flex flex-column\\">
    <div class=\\"d-flex flex-row justify-content-center font-weight-bold mt-2\\">
      <choppalogoatom-stub css=\\"h1\\"></choppalogoatom-stub>
    </div>
    <div class=\\"d-flex flex-row\\">
      <h1 class=\\"h5 justify-content-center mt-2 mb-0\\">
        404 - You've taken the wrong flight!
      </h1>
    </div>
    <div class=\\"d-flex flex-row justify-content-center\\">
      <router-link-stub to=\\"/\\" tag=\\"a\\" ariacurrentvalue=\\"page\\" event=\\"click\\" class=\\"d-flex mt-1\\">
        Fly home
      </router-link-stub>
    </div>
  </div>
</div>"
`;
